CHEWBBACA/CHEWBBACA_NS/send2NS_refactored.py

```diff
@@ -0,0 +1,237 @@
+                i += 1
+                print (row[0] + " discarded from further analysis because missing data count is over " + str(percentMDallowed))
+                listGenomes2discard.append(row[0])
+                continue
+
+#            hasAlleles = False
+
+            #if number of missing data is acceptable, get the alleles that are only local and store them on a dictionary {locus:[new alleles]}
+            while i<len(row):
+
+                gene = headers[i]
+                if "*" in row[i]:
+                    new_allele= int(row[i].replace('*',''))
+                    dict_new_alleles[headers[i]].append(new_allele)
+
+                i+=1
+
+    lists2print = []
+    modifiedProfileDict = {}
+    i = 0
+
+    #create the string of the new profile
+    while i < len(profileDict["FILE"]):
+        aux=[]
+        for header in listHeaders:
+            aux.append(profileDict[header][i])
+        i += 1
+        modifiedProfileDict[aux.pop(0)] = aux
+        lists2print.append(aux)
+
+    newProfileStr=('\t'.join(map(str,listHeaders))) + "\n"
+    i = 0
+    for elem in lists2print:
+        newProfileStr += str(list(genomes)[i]) + "\t" + ('\t'.join(map(str, elem))) + "\n"
+        i += 1
+        
+    for genome in sorted(modifiedProfileDict):
+
+        if genome in listGenomes2discard:
+            print (genome + " profile has a low locus count, probably a different species")
+            continue
+        
+        else:
+            profile = modifiedProfileDict[genome]
+            ola = process_genome(genome, schemaUri, profile, listHeaders, headers_post)
+                
+    print ("DONE")
+
+
+
+
+
+def get_schema_vars(schemapath):
+
+    #get schema .config file and its variables, schema uri and last server data
+    schemapath_config=os.path.join(schemapath,".config.txt")
+    try:
+        with open(schemapath_config) as fp:
+            lines=[]
+            for line in fp:
+                lines.append(line.strip())
+
+    except:
+        print ("your schema has no config file")
+        raise
+
+    lastSyncServerDate=lines[0]
+    schemaUri=lines[1]
+
+    return lastSyncServerDate,schemaUri
+
+
+def main(profileFile,pathSchema,token,metadataFile,percentMDallowed,cpu):
+    
+    cpu2Use=int(cpu)
+    percentMDallowed = 0.5
+    percentMDallowed=float(percentMDallowed)
+    # profileFile = "/home/pcerqueira/Lab_Software/refactored_ns/ns_security/extra_scripts/toxic_schema_test/results_alleles_teste.tsv"
+    # schemaUri = "http://127.0.0.1:5000/NS/api/species/13/schemas/6"
+
+    #get schema uri and date from schema .config file
+    lastSyncServerDate, schemaUri = get_schema_vars(pathSchema)
+    
+    server_url = schemaUri.split("/species")[0]
+
+    
+    ## FOR DEBUG ONLY! ###########################################################################
+    # Log the user in
+    token = ut.login_user_to_NS(server_url, "test@refns.com", "mega_secret")
```

Comment:
you want to add this to the PR???

Reply:
File in development. Removed from PR and highlighted hard-coded credentials that are used in development.